go/vt/vtgate/evalengine/expr_bvar.go

```diff
@@ -83,9 +83,6 @@ func (bv *BindVariable) eval(env *ExpressionEnv) (eval, error) {
 			return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "query argument '%s' cannot be a tuple", bv.Key)
 		}
 		typ := bvar.Type
-		if bv.typed() {
-			typ = bv.Type
-		}
```

Comment:
Hmm, not sure why this could be removed?

Reply:
so, it looks like if we keep this, we will sometimes try to cast a `null` value to an int64, and we get this error:

```
cannot parse int64 from empty string (errno 1105) (sqlstate HY000) during query: select sum(tbl1.a), min(tbl0.b) from t10 as tbl0, t10 as tbl1 left join t10 as tbl2 on tbl1.a = tbl2.a and tbl1.b = tbl2.k
```

I think that is what happens, but I'm not sure about the reasoning.